doc/neps/nep-0000.rst

```diff
@@ -0,0 +1,219 @@
+documenting dissenting opinions.
+
+Because the NEPs are maintained as text files in a versioned
+repository, their revision history is the historical record of the
+feature proposal [1]_.
+
+
+Types
+^^^^^
+
+There are three kinds of NEP:
+
+1. A **Standards Track** NEP describes a new feature or implementation
+   for NumPy.
+
+2. An **Informational** NEP describes a NumPy design issue, or
+   provides general guidelines or information to the NumPy community,
+   but does not propose a new feature.  Informational NEPs do not
+   necessarily represent a NumPy community consensus or
+   recommendation, so users and implementers are free to ignore
+   Informational NEPs or follow their advice.
+
+3. A **Process** NEP describes a process surrounding NumPy, or
+   proposes a change to (or an event in) a process.  Process NEPs are
+   like Standards Track NEPs but apply to areas other than the NumPy
+   language itself.  They may propose an implementation, but not to
+   NumPy's codebase; they often require community consensus; unlike
+   Informational NEPs, they are more than recommendations, and users
+   are typically not free to ignore them.  Examples include
+   procedures, guidelines, changes to the decision-making process, and
+   changes to the tools or environment used in NumPy development.
+   Any meta-NEP is also considered a Process NEP.
+
+
+NEP Workflow
+------------
+
+The NEP process begins with a new idea for NumPy.  It is highly
+recommended that a single NEP contain a single key proposal or new
+idea. Small enhancements or patches often don't need
+a NEP and can be injected into the NumPy development workflow with a
+pull request to the NumPy `repo`_. The more focused the
+NEP, the more successful it tends to be.
+If in doubt, split your NEP into several well-focused ones.
+
+Each NEP must have a champion---someone who writes the NEP using the style
+and format described below, shepherds the discussions in the appropriate
+forums, and attempts to build community consensus around the idea.  The NEP
+champion (a.k.a. Author) should first attempt to ascertain whether the idea is
+NEP-able. Posting to the numpy-discussion `mailing list`_ is the best way to go
+about doing this.
+
+Following a discussion on the mailing list, the proposal should be submitted as a
+draft NEP via a `GitHub pull request`_ to the ``doc/neps`` directory with the name
+``nep-<n>.rst`` where ``<n>`` is an appropriately assigned number. The draft must
+use the :doc:`nep-template` file. Once a formal proposal has been submitted as a
+PR, it should be announced on the mailing list.
+
+Standards Track NEPs consist of two parts, a design document and a
+reference implementation.  It is generally recommended that at least a
+prototype implementation be co-developed with the NEP, as ideas that sound
+good in principle sometimes turn out to be impractical when subjected to the
+test of implementation.
```

Comment:
Where should the code be stored?

Reply:
I added some additional text.  Feel free to suggest further changes.